airflow/api_fastapi/core_api/routes/ui/grid.py

```diff
@@ -0,0 +1,324 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import collections
+import itertools
+import operator
+from functools import cache
+
+from fastapi import HTTPException, Request, status
+from sqlalchemy import func, select
+from sqlalchemy.sql.operators import ColumnOperators
+from typing_extensions import Any
+
+from airflow import DAG
+from airflow.api_fastapi.common.db.common import SessionDep, paginated_select
+from airflow.api_fastapi.common.parameters import (
+    OptionalDateTimeQuery,
+    QueryDagRunRunTypesFilter,
+    QueryDagRunStateFilter,
+    QueryLimit,
+    QueryOffset,
+    SortParam,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.datamodels.ui.grid import (
+    GridDAGRunwithTIs,
+    GridResponse,
+    GridTaskInstanceSummary,
+)
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.configuration import conf
+from airflow.exceptions import AirflowConfigException
+from airflow.models import DagRun, MappedOperator, TaskInstance
+from airflow.models.baseoperator import BaseOperator
+from airflow.models.taskmap import TaskMap
+from airflow.utils import timezone
+from airflow.utils.state import TaskInstanceState
+from airflow.utils.task_group import MappedTaskGroup, TaskGroup
+
+grid_router = AirflowRouter(prefix="/grid", tags=["Grid"])
+
+
+@grid_router.get(
+    "/{dag_id}",
+    include_in_schema=False,
+    responses=create_openapi_http_exception_doc([status.HTTP_400_BAD_REQUEST, status.HTTP_404_NOT_FOUND]),
+)
+def grid_data(
+    dag_id: str,
+    run_types: QueryDagRunRunTypesFilter,
+    run_states: QueryDagRunStateFilter,
+    session: SessionDep,
+    offset: QueryOffset,
+    request: Request,
+    num_runs: QueryLimit,
+    base_date: OptionalDateTimeQuery = None,
+    root: str | None = None,
+    filter_upstream: bool = False,
+    filter_downstream: bool = False,
```

Comment:
I renamed those for the `graph_data` to `include_upstream`, and `include_downstream` which I think are more intuitive.

Here `filter_upstream=False` will actually `include_upstream=False` => no upstream. (Where we asked for not filtering upstream).

Reply:
Also we could move those to a `ui.parameters.py` file. Because they are shared accross multiple  ui endpoints. (same for root). And default value is the same.

Reply:
+1 on moving to a shared param since the grid and graph data endpoints will be used together in the UI so often.

Reply:
Moved into parameters.py